applications/asset_tracker/Kconfig

```diff
@@ -217,8 +217,13 @@ endmenu # Motion
 menu "Cloud"
 
 config MQTT_KEEPALIVE
-	int
-	default 1200
+	int "Time after last transmission to send a ping to keep connection on"
+	default 120
+	help
+	  This should be set to the expected time between messages sent from
+	  this device. Making it larger than the expected data period does not
+	  conserve bandwidth, as the ping is only sent if nothing else
+	  has been within the specified period.
```

Comment:
I'm not sure the information belongs here, but we should somehow make users aware that sending a ping interrupts GPS search for an unknown time, typically ranging from 5 to 70 seceonds, depending on network configs out of out control. So with a keep-alive of 120 seconds might result in GPS search periods of 50 seconds max, which could increase TTFF rather significantly.

Reply:
Good point.  Not sure either.

Reply:
How about by adding something to README.rst, in the power optimization section, together with similar comments as the Kconfig change about keepalive?

Reply:
I think that the keepalive time should be set based on what is needed to keep the MQTT connection alive. Currently the maximum MQTT keepalive time for AWS is 1200s, i.e. the value we currently have in Asset Tracker by default. This not only affects the sending of MQTT pings, but also the time when the broker determines that the connection is dead if it doesn't receive anything from the client. Usually this happens when 1.5 times the keepalive has passed, so setting this to 120 seconds would mean that if the device can't get a ping through in 60 seconds, the broker will drop the MQTT connection. This is something that can happen, especially with LTE-NB.

Also when Asset Tracker is performing GPS search, it stops sending data to the cloud so that LTE doesn't block GPS. As pointed out by @jtguggedal, MQTT pings would still be sent during GPS search and would affect the GPS performance. Also because of this it would be good to keep the keepalive as long as possible.

It could be a good idea to mention that NAT timeouts can cause the TCP connection to be dropped and that can be worked around by setting the MQTT keepalive time to a shorter value. But by default I think we should keep it as long as possible and my proposal is to keep the current default 1200s.

Reply:
I have added a jira task TG91-236 that is to not rely on default configs on parameters that typically should be tuned when customters start to do optimizations. I think the defaul can stay at 1200 and we add the config in the .conf files that set it to 120 for the best "out of the box"-experience

Reply:
@jhn-nordic so does my PR get approved, and then another to address TG91-236?

Reply:
I think we should keep this at 1200, but the help text changes are good. Perhaps also add a sentence about NAT here, as this is the config users should tweak if they experience issues with that.

Reply:
Agree with @jtguggedal. Set this to 1200s again and I can approve

Reply:
The whole reason I wanted to change this was to improve out-of-box experience, especially when having trouble getting a GPS fix.  Ironically this change to 120 will make GPS fixes take longer, but will also prevent NAT timeouts and unresponsive devices for ~20 minutes after a failure to get a fix, so overall it is a win in my opinion.

I have gone ahead and changed the Kconfig value back to 1200, added more info about NAT and GPS effects of shorter values to the help text, and changed the prj.conf and prj_nrf9160_pca20035ns.conf values to 120 seconds to improve OOB.